h2/src/main/org/h2/server/pg/PgServerThread.java

```diff
@@ -734,6 +739,57 @@ private void writeDataColumn(Value v, int pgType, boolean text) throws IOExcepti
         }
     }
 
+    private static final int[] POWERS10 = {1, 10, 100, 1000, 10000};
+    private static final int MAX_GROUP_SCALE = 4;
+    private static final int MAX_GROUP_SIZE = POWERS10[4];
+
+    private static int divide(BigInteger[] unscaled, int divisor) {
+        BigInteger[] bi = unscaled[0].divideAndRemainder(BigInteger.valueOf(divisor));
+        unscaled[0] = bi[0];
+        return bi[1].shortValue();
+    }
+
+    // https://www.npgsql.org/dev/types.html
+    // https://github.com/npgsql/npgsql/blob/main/src/Npgsql/TypeHandlers/NumericHandlers/NumericHandler.cs#L167
+    private void writeNumericBinary(BigDecimal value) throws IOException {
+        int weight = 0;
+        List<Integer> groups = new ArrayList<>();
+        int scale = value.scale();
+        int signum = value.signum();
+        if (signum != 0) {
+            BigInteger[] unscaled = new BigInteger[] {value.unscaledValue()};
+            if (signum < 0) {
+                unscaled[0] = unscaled[0].negate();
+            }
+            weight = -scale / MAX_GROUP_SCALE - 1;
+            int remainder = 0;
+            int scaleChunk = scale % MAX_GROUP_SCALE;
+            if (scaleChunk > 0) {
+                remainder = divide(unscaled, POWERS10[scaleChunk]) * POWERS10[MAX_GROUP_SCALE - scaleChunk];
+                if (remainder != 0) {
+                    weight--;
+                }
+            }
+            if (remainder == 0) {
+                while ((remainder = divide(unscaled, MAX_GROUP_SIZE)) == 0) {
+                    weight++;
+                }
+            }
+            groups.add(remainder);
+            while (unscaled[0].signum() != 0) {
+                groups.add(divide(unscaled, MAX_GROUP_SIZE));
+            }
+        }
+        writeInt(8 + groups.size() * 2);
+        writeShort(groups.size());
+        writeShort(groups.size() + weight);
+        writeShort(signum < 0 ? 16384 : 0);
+        writeShort(scale);
```

Comment:
And here we have another problem with scale. `SELECT 1E-100000` and `SELECT 1E-60000` throw
```
Exception in thread "main" java.lang.IllegalArgumentException: invalid scale in "numeric" value
	at org.postgresql.util.ByteConverter.numeric(ByteConverter.java:183)
	at org.postgresql.util.ByteConverter.numeric(ByteConverter.java:147)
	at org.postgresql.jdbc.PgResultSet.getNumeric(PgResultSet.java:2355)
	at org.postgresql.jdbc.PgResultSet.getBigDecimal(PgResultSet.java:2332)
	at org.postgresql.jdbc.PgResultSet.getBigDecimal(PgResultSet.java:390)
```
`SELECT 1E-70000` returns `0E-4464`.

I think behavior should be consistent, for example, scale can be normalized to the supported by PosgreSQL range.